mod-source-record-storage-server/src/main/resources/liquibase/tenant/scripts/v-5.10.0/2024-12-10--20-00-create-delete-old-marc-indexers-versions.xml

```diff
@@ -31,45 +31,56 @@
     </sql>
   </changeSet>
 
-  <changeSet id="2024-12-10--17-00-create-delete-old-marc-indexers-versions" author="Volodymyr_Rohach">
+  <changeSet id="2024-12-10--17-00-create-delete-old-marc-indexers-versions" author="Aliaksandr_Fedasiuk">
+    <sql>
+      DROP PROCEDURE IF EXISTS ${database.defaultSchemaName}.delete_old_marc_indexers_versions;
+    </sql>
     <createProcedure>
-      CREATE OR REPLACE PROCEDURE ${database.defaultSchemaName}.delete_old_marc_indexers_versions() LANGUAGE plpgsql AS $$
+      CREATE OR REPLACE PROCEDURE ${database.defaultSchemaName}.delete_old_marc_indexers_versions(IN pLimit INTEGER) LANGUAGE plpgsql AS $$
       BEGIN
-      WITH dirty_records AS (
-        SELECT mrt.marc_id, mrt.version
-        FROM marc_records_tracking mrt
-        WHERE mrt.is_dirty = true
-      ),
-      deleted_dirty AS (
-      DELETE FROM marc_indexers mi
-        USING dirty_records dr
-      WHERE mi.marc_id = dr.marc_id AND dr.version > mi.version
-        RETURNING mi.marc_id
+        WITH dirty_records AS (
```

Comment:
please add `WITH dirty_records AS materialized`. This will allow postgres to compute the CTE before using it elsewhere. It will let the postgres optimizer know if the number of rows in the CTE. When the CTE has 0 rows, a more appropriate query plan will be used with the marc_indexers table.

Reply:
completed, thank you